csharp/extractor/Semmle.Extraction.CSharp/Entities/TypeMention.cs

```diff
@@ -23,27 +23,36 @@ private TypeMention(Context cx, TypeSyntax syntax, IEntity parent, Type type, Mi
             this.loc = loc;
         }
 
-        private static TypeSyntax GetElementType(TypeSyntax type)
+        private TypeSyntax GetArrayElementType(TypeSyntax type)
```

Comment:
Perhaps `GetUnderlyingType` is a better name?

Reply:
I thought about this too. This method should not be called from `case SyntaxKind.NullableType` in `Populate`. It should only be used to get the element type of array-like `TypeSyntax` elements (array, pointer), hence the name. However, it's recursive and during the recursion, we need to also handle `NullableTypeSyntax`, and this introduces a bit of confusion.